src/aap_eda/conf/registry.py

```diff
@@ -0,0 +1,190 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from collections import OrderedDict
+from dataclasses import dataclass
+from typing import Any
+
+import yaml
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import transaction
+
+from aap_eda.core.models import Setting
+from aap_eda.core.utils.crypto.base import SecretValue
+
+ENCRYPTED_STRING = "$encrypted$"
+
+
+class InvalidKeyError(Exception):
+    ...
+
+
+class InvalidValueError(Exception):
+    ...
+
+
+@dataclass
+class RegistryData(object):
+    name: str
+    default: Any
+    type: type = str
+    read_only: bool = False
+    encrypted: bool = False
```

Comment:
since all the values are actually encrypted, I feel this name confusing. Would not be better to name "is_secret" ?